src/Lachain.Core/Blockchain/Operations/BlockManager.cs

```diff
@@ -150,7 +152,36 @@ public Block LatestBlock()
         [MethodImpl(MethodImplOptions.Synchronized)]
         public Block? GetByHeight(ulong blockHeight)
         {
-            return _stateManager.LastApprovedSnapshot.Blocks.GetBlockByHeight(blockHeight);
+            try
+            {
+                /* default cache size limit is 100 items */
+                var sizeLimit = 100;
```

Comment:
It is better to read cache limit only once in constructor instead of reading it every time block is requested

Reply:
Okay